src/session.rs

```diff
@@ -217,40 +263,70 @@ impl Session {
         }
     }
 
-    pub fn cookies(&self) -> Vec<cookie_crate::Cookie> {
-        let mut cookies = vec![];
-        for (name, c) in &self.content.cookies {
-            let mut cookie_builder = cookie_crate::Cookie::build(name, &c.value);
-            if let Some(expires) = c.expires {
-                cookie_builder =
-                    cookie_builder.expires(time::OffsetDateTime::from_unix_timestamp(expires));
-            }
-            if let Some(ref path) = c.path {
-                cookie_builder = cookie_builder.path(path);
-            }
-            if let Some(secure) = c.secure {
-                cookie_builder = cookie_builder.secure(secure);
-            }
-            cookies.push(cookie_builder.finish());
+    pub fn cookies(&self) -> impl Iterator<Item = Result<cookie_store::Cookie<'static>>> + '_ {
+        match &self.content.cookies {
+            Cookies::Map(_) => unreachable!(),
+            Cookies::List(cookies) => cookies.iter().map(|cookie| {
+                let mut cookie_builder =
+                    cookie_store::RawCookie::build(cookie.name.clone(), cookie.value.clone());
+
+                if let Some(expires) = cookie.expires {
+                    cookie_builder =
+                        cookie_builder.expires(time::OffsetDateTime::from_unix_timestamp(expires)?);
+                }
+                if let Some(path) = &cookie.path {
+                    cookie_builder = cookie_builder.path(path.clone());
+                }
+                if let Some(secure) = cookie.secure {
+                    cookie_builder = cookie_builder.secure(secure);
+                }
+
+                let mut cookie_url = self.url.clone();
+                if let Some(domain) = &cookie.domain {
+                    cookie_url = format!("http://{domain}").parse()?;
```

Comment:
[`url::Host::parse`](https://docs.rs/url/latest/url/enum.Host.html#method.parse) looks a little easier

Reply:
That would return a `Host` but I need to construct a `cookie_url` that matches the cookie's domain. This is to avoid any `DomainMismatch` errors later when calling `try_from_raw_cookie()`.